ios/browser/api/ads/brave_ads.mm

```diff
@@ -651,11 +651,13 @@ - (void)reportPromotedContentAdEvent:(NSString*)uuid
       static_cast<ads::mojom::PromotedContentAdEventType>(eventType));
 }
 
-- (void)purgeOrphanedAdEvents:(AdsAdType)adType {
+- (void)purgeOrphanedAdEvents:(AdsAdType)adType
+                   completion:(void (^)(BOOL success))completion {
   if (![self isAdsServiceRunning]) {
     return;
   }
-  ads->PurgeOrphanedAdEventsForType(static_cast<ads::mojom::AdType>(adType));
+  ads->PurgeOrphanedAdEventsForType(static_cast<ads::mojom::AdType>(adType),
+                                    completion);
```

Comment:
This won't compile because `PurgeOrphanedAdEventsForType` expects a `base::OnceCallback`.  Likely you just need to wrap the completion in `base::BindOnce` (i.e. `base::BindOnce(completion)`)

Reply:
@kylehickinson, IOS build is compiled actually, because `Ads::PurgeOrphanedAdEventsForType` expects `std::function<void(const bool)>` declared here: https://github.com/brave/brave-core/pull/11945/files#diff-c73bc533ece7d4b70d5b88529ae50c6aa7c792df81b14923289a3648157f3614R35
using of `std::function` will be refactored here: https://github.com/brave/brave-browser/issues/20940